lib/tapioca/compilers/dsl_compiler.rb

```diff
@@ -76,6 +76,19 @@ def run(&blk)
         result.compact
       end
 
+      sig { params(generator_name: String).returns(T::Boolean) }
+      def generator_enabled?(generator_name)
+        # Try to find built-in tapioca generator first, then globally defined generator. The
+        # explicit `break` ensures the class is returned, not the `potential_name`.
+        generator_klass = ["Tapioca::Compilers::Dsl::#{generator_name}", generator_name].find do |potential_name|
+          break Object.const_get(potential_name)
+        rescue NameError
+          # Skip if we can't find generator by the potential name
+        end
```

Comment:
Should we extract this so it can be share with `constantize_generators`? https://github.com/Shopify/tapioca/blob/d63482a222f0cd5af4e8a765bcc6b9bfa801ec1a/lib/tapioca/generators/dsl.rb#L191-L197

Reply:
Done!